include/caffe/net.hpp

```diff
@@ -192,7 +192,8 @@ class Net {
   /// @brief Append a new bottom blob to the net.
   int AppendBottom(const NetParameter& param, const int layer_id,
                    const int bottom_id, set<string>* available_blobs,
-                   map<string, int>* blob_name_to_idx);
+                   map<string, int>* blob_name_to_idx,
+                   bool skip_propagate = false);
```

Comment:
There's no reason to make `skip_propagate` a new input -- just get it from `layer_param` in `AppendBottom`. (I don't think there should be any changes to `Net::Init` if you do it this way, but I might have overlooked something.)